### ISSUE TITLE
unique invoice number

### ISSUE THREAD

**Issue by sanubari97** [opensourcepos/opensourcepos#1678]
Got error when input "sales by invoice" said "Please enter an unique invoice number"

?

**Comment by SteveIreland** [comment]
This is caused by a problem with your data or a configuration issue.  If you cannot figure it out for yourself and need some free support or sympathy please provide the requested environment information and some details about how you are configured related to invoices

**Comment by jekkos** [comment]
Probably your invoice counter is not setup correctly or you changed it yourself? It tries to complete an invoice with number that is already in the db

**Comment by sanubari97** [comment]
![untitled](https://user-images.githubusercontent.com/30258590/31594061-411562aa-b265-11e7-8e75-b4f7b18f6f03.jpg)
this is my config,
I deleted some invoice sales from the "takings", is that the cause ? then how do I fix it ?

**Comment by SteveIreland** [comment]
Yes, the {CO} token assigns an invoice number representing the current count of records in the `ospos_sales` table.  Since you deleted some records that throws the count off.

If you want a quick fix then change the invoice format to  `{ISEQ}` and set the last used invoice number to the last one that was actually assigned as the invoice.
